### PR TITLE
Remove the custom prefix from the table name length when checking on …

### LINKED ISSUE #15229: Update to 16.0 - InvalidArgumentException: Primary index name on "xxxxxx_collres_collections" is too long.
Getting following error while updating from 15 to 16 with Web - updater:

Aktualisierung auf 16.0.0

InvalidArgumentException: Primary index name on "xxxxxxxx_collres_collections" is too long.
Detaillierte Protokollmeldungen

Update vorbereiten

Log-Level auf "debug" gesetzt

Wartungsmodus eingeschaltet

Reparaturschritt: Repair MySQL collation

Reparaturinformation: All tables already have the correct collation -> nothing to do

Reparaturschritt: Repair SQLite autoincrement

Reparaturschritt: Copy data from accounts table when migrating from ownCloud

Reparaturschritt: Drop account terms table when migrating from ownCloud

Das Datenbankschema wird aktualisiert

InvalidArgumentException: Primary index name on "xxxxxxxx_collres_collections" is too long.

### PR DESCRIPTION
**Pull request by nickvergessen**
…the default primary index

Fix #15229

### CODE REVIEW

**Comment by skjnldsv** [comment]
Well, we should probably also check and/or enforce `innodb_large_prefix` as well as mariadb > 12.2.
Just upgrading mariadb fixed everything for me.

**Review by skjnldsv — APPROVED** [review]
But otherwise, fine forme

**Comment by nickvergessen** [comment]
> Well, we should probably also check and/or enforce innodb_large_prefix as well as mariadb > 12.2.

That is completely unrelated. But feel free to create a new issue, explain what the problem is and assign it to me.

**Comment by skjnldsv** [comment]
> That is completely unrelated. But feel free to create a new issue, explain what the problem is and assign it to me.

Ah sorry, I thought it was related to the `Syntax error or access violation: 1071 Specified key was too long; max key length is 767 bytes`

File: lib/private/DB/MigrationService.php [hunk 1]
```diff
@@ -547,7 +547,7 @@ public function ensureOracleIdentifierLengthLimit(Schema $sourceSchema, Schema $
 				if (!$isUsingDefaultName && \strlen($indexName) > 30) {
 					throw new \InvalidArgumentException('Primary index name  on "'  . $table->getName() . '" is too long.');
 				}
-				if ($isUsingDefaultName && \strlen($table->getName()) > 26) {
+				if ($isUsingDefaultName && \strlen($table->getName()) - $prefixLength > 22) {
```
Comment (blizzz) [inline lib/private/DB/MigrationService.php]: not 23?

Comment (nickvergessen) [reply lib/private/DB/MigrationService.php]: 30 - 3 (`oc_`) - 5 (`_pkey`) it was off by one before... math problems :dancer:

Comment (nickvergessen) [reply lib/private/DB/MigrationService.php]: Ah, you are right, oracle is `_seq` so 4 only. let me fix it

**Review by blizzz — APPROVED** [review]
makes sense

**Pull request merged**